numpy/core/src/multiarray/getset.c

```diff
@@ -332,6 +328,13 @@ array_data_set(PyArrayObject *self, PyObject *op)
     Py_ssize_t buf_len;
     int writeable=1;
 
+    int ret = DEPRECATE("Assigning the 'data' attribute is an "
```

Comment:
Sure thing.  I think my very first PR to `numpy` might have that!  Let me double check.